src/utils.c

```diff
@@ -118,14 +119,31 @@ ts_time_value_to_internal(Datum time_val, Oid type_oid)
 {
 	Datum res, tz;
 
+	if (TS_TIME_IS_INTEGER_TIME(type_oid))
+	{
+		/* Integer time types have no distinction between min, max and
+		 * infinity. We don't want min and max to be turned into infinity for
+		 * these types so check for those values first. */
+		if (TS_TIME_DATUM_IS_MIN(time_val, type_oid))
+			return ts_time_get_min(type_oid);
+
+		if (TS_TIME_DATUM_IS_END(time_val, type_oid))
+			return ts_time_get_end(type_oid);
+	}
```

Comment:
This seems to cover this case, but I am not sure if there are others. It would be a lot safer to ensure that this check would not be necessary, that is, ensure that the "end" and "min" values are distinct also for integer types.

Reply:
> This seems to cover this case, but I am not sure if there are others. It would be a lot safer to ensure that this check would not be necessary, that is, ensure that the "end" and "min" values are distinct also for integer types.

That's unfortunately impossible with how things currently work. It would require reducing the supported/legal values in order to reserve space for "infinity".

Reply:
I realize that, and the PR stays approved so you can push if you want, but I think we need to consider if we can improve this in some way in the future.

Reply:
I am open to suggestions, but I don't really see how we can improve without either (1) reserving values, or (2) handling every time type separately at every place in the code that needs it.

Reply:
I think it requires a rewrite. I think this PR improves the current situation, so I think we should handle the necessary changes as a separate PR.